commands/blackjack.js

```diff
@@ -0,0 +1,199 @@
+      }
+    }
+  }
+
+  return dealersCardSum;
+};
+
+const handleStay = (interaction, handData) => {
+  let handDataParse = handData.split('-');
+  let dealersCards = [handDataParse[1], handDataParse[2]];
+  let playersCards = '';
+  let playersCardSum = parseInt(handDataParse[handDataParse.length - 1]);
+
+  // parse player hand
+  for (var i = 3; i < handDataParse.length - 1; i++) {
+    playersCards = playersCards.concat(handDataParse[i], ', ');
+  }
+
+  let dealersCardSum = playDealersHand(dealersCards);
+  
+  let resultContent;
+  if (dealersCardSum < playersCardSum || dealersCardSum > 21) {
+    resultContent = 'PLAYER WINS!'
+  } else if (dealersCardSum === playersCardSum) {
+    resultContent = 'YALL PUSHED!';
+  } else {
+    resultContent = 'DEALER WINS!';
+  }
+  interaction.update({
+    content: `**Playing Blackjack**\n\nYour Hand: ${playersCards} \nTotal: ${playersCardSum} \n\nDealer's Hand: ${dealersCards} \nDealer's Total: ${dealersCardSum}\n\n**Result: ${resultContent}**`,
+    components: [],
+  });
+};
+
+const handleHit = (interaction, handData) => {
+  let handDataParse = handData.split('-');
+  let cards = [];
+  let cardSum = parseInt(handDataParse[handDataParse.length - 1]);
+  let handContent = '';
+
+  for (var i = 3; i < handDataParse.length - 1; i++) {
+    // parse player hand
+    if (i !== 1 && i !== 2) {
+      cards.push(handDataParse[i]);
+      handContent = handContent.concat(handDataParse[i], ', ');
+    }
+  }
+
+  const deck = new Deck(deckOf52Cards);
+  deck.shuffle();
+
+  // draw the next card
+  const card = deck.draw(1);
+  const cardValue = getCardValue(card);
+  cardSum = cardSum + cardValue;
+  cards.push(card);
+
+  // player bust
+  if (cardSum > 21) {
+    // does the player have an Ace?
+    [cards, cardSum] = convertAceToOne(cards, cardSum);
+    // player still bust
+    if (cardSum > 21) {
+      handContent = handContent.concat(card);
+      interaction.update({
+        content: `**Playing Blackjack**\n\nYour Hand: ${handContent} \nTotal: ${cardSum} \n\n**Sorry, you BUSTED**`,
+        components: [],
+      });
+      return;
+    }
+  }
+
+  let hitCustomIdKey = 'hit-';
+  let stayCustomIdKey = 'stay-';
+  handContent = '';
+
+  // add dealers cards to customId
+  for (var i = 1; i < 3; i++) {
+    hitCustomIdKey = hitCustomIdKey.concat(handDataParse[i], '-');
+    stayCustomIdKey = stayCustomIdKey.concat(handDataParse[i], '-');
+  }
+
+  // add players cards to customId and handContent
+  for (var i = 0; i < cards.length; i++) {
+    hitCustomIdKey = hitCustomIdKey.concat(cards[i], '-');
+    stayCustomIdKey = stayCustomIdKey.concat(cards[i], '-');
+    handContent = handContent.concat(cards[i], ', ');
+  }
+
+  hitCustomIdKey = hitCustomIdKey.concat(cardSum);
+  stayCustomIdKey = stayCustomIdKey.concat(cardSum);
+
+  const row = new ActionRowBuilder().addComponents([
+    new ButtonBuilder().setCustomId(hitCustomIdKey).setLabel('Hit').setStyle('Primary'),
```

Comment:
could be easier to use a modifiable json file to just store this info

Reply:
Hmm probably could do that but reading and writing to a file seems unnecessarily complex if we don't need to maintain the state outside of the interaction. By doing it this way, it allows us to keep the state within the interaction.